src/internal.rs

```diff
@@ -421,28 +419,66 @@ impl Rav1dTask {
             ..Default::default()
         }
     }
+}
 
-    pub fn next(&self) -> Rav1dTaskIndex {
-        self.next.load(Ordering::SeqCst)
-    }
-
-    pub fn set_next(&self, next: Rav1dTaskIndex) {
-        self.next.store(next, Ordering::SeqCst)
+impl PartialOrd for Rav1dTask {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
     }
 }
 
-impl Rav1dTask {
-    pub fn without_next(&self) -> Self {
-        Self {
-            frame_idx: self.frame_idx,
-            tile_idx: self.tile_idx,
-            type_0: self.type_0,
-            sby: self.sby,
-            recon_progress: self.recon_progress,
-            deblock_progress: self.deblock_progress,
-            deps_skip: self.deps_skip.clone(),
-            next: Default::default(),
+impl Ord for Rav1dTask {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        // We want `sort` to put tasks in priority order.
+        // To do that, we return:
+        //  - `Less`    if `self` is of higher priority than `other`
+        //  - `Greater` if `self` is of lower priority than `other`
+        // This requires that there are no Init, InitCdf or film grain tasks.
```

Comment:
Could this be a doc comment on the function?

Reply:
Folded into the new https://github.com/memorysafety/rav1d/pull/1447/commits/3a02af0d171821917612e20705d80ee8bbad6623